### PR TITLE
openEO Federation

### PR DESCRIPTION
**Pull request by jdries**
Initial pages for openEO federation

### CODE REVIEW

**Review by JanssenBrm — CHANGES_REQUESTED** [review]
Reviewed the text and made some editorial changes. However, I've added some suggestions for improvements in the comments and some references seem missing.

File: APIs/openEO/federation/openeo_federation.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+https://openeofed.dataspace.copernicus.eu
+
+This capability is currently in a phase where we encourage early adopters and projects with specific needs to try it out.
+It has already been extensively tested by users of other projects, but the configuration in the CDSE is new and thus we 
+offer it as a 'beta' service at first, allowing us to still make changes to the configuration based on user feedback.
+
+When using the federated endpoint, you will be asked to give consent for basic user information to be shared with the other providers
+that make up the federation, as well as to accept the terms and conditions. These external providers all comply with the 
+regulations of the federation, but are not necessarily part of the CDSE service. You can withdraw your consent at any time, via
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/openeo_federation.md]: yes, this is a crucial but difficult part I believe. Need to find some better wording that is understandable and legally sound. Maybe just need to expand on it rather than trying to fit it into a single sentence.

File: APIs/openEO/federation/openeo_federation.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+## Getting help
+
+We encourage users to use the CDSE forum to ask questions about the federation. The CDSE support team is not required to
+help with issues related to non-CDSE openEO services, so the forum is a place where service providers and users can meet.
+When using free credits offered by the service provider, the support is not considered to be included. The CDSE openEO endpoint 
+does include support under the normal conditions.
+
+## Long term continuity
+
+Members in the federation may not have the same long term guarantees as the CDSE service. Members may discontinue their
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/openeo_federation.md]: good point, we call them members/backends/providers

File: APIs/openEO/federation/backends/api.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+The requirement to implement two of *L1A*, *L1B*, and *L1C* has been restricted for openEO Platform to always require *L1A: Synchronous Processing* and *L1B: Batch Jobs* (see req. no. 703). This means that *L1C: Secondary Web Services* is optional.
+
+<figure>
+    <img src="./profiles/api.png" alt="The hierarchy of openEO and openEO Platform API profiles.">
+    <figcaption>An overview of the openEO and openEO Platform API profiles.</figcaption>
+</figure>
+
+### LP: Required for openEO Platform
+
+The profile only lists requirements that are not covered by the openEO profile *L2: Recommended* yet.
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/backends/api.md:20]: indeed, your interpretation is correct. Here we also still need to adjust reference to platform.

File: APIs/openEO/federation/backends/collections.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+3. Collections without an end time are assumed to be active missions. By default, 99% of items in these collections should be available within 48 hours after being published by the producer. This gives users a basic guarantee with respect to timeliness of products. 
+4. Collection metadata should be valid STAC metadata and must include all extensions in `stac_extensions`. Tools such as [STAC-validator](https://github.com/stac-utils/stac-validator) can indicate obvious issues.
+5. FAIR principle R1: [(Meta)data are richly described with a plurality of accurate and relevant attributes](https://www.go-fair.org/fair-principles/r1-metadata-richly-described-plurality-accurate-relevant-attributes/)
+6. Collections have to follow harmonization guidelines specified below, if applicable.
+7. Collections naming (id, dimensions, bands) should remain constant. 
+8. Backwards incompatible changes or removal need to be announced with a lead time of 6 months, together with a migration path.
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/backends/collections.md:53]: The need to announce it is already in there?

File: APIs/openEO/federation/backends/index.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+- [Processes](./processes.md)
+
+## Benefits of joining the federation
+
+This federated concept provides a unique opportunity for smaller organizations to join strengths, allowing to build an
+offer that matches or even exceeds capabilities of well-known large scale cloud providers. This is a key element for long
+term sustainability of participants. There are however also short term benefits: 
+
+1. **Increased visibility**: the federation is a single endpoint for users to access multiple services, increasing the visibility of your service.
+2. **Increased user base**: users of the federation can access your service without the need to register, increasing the potential user base.
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/backends/index.md:42]: fine for me!

File: APIs/openEO/federation/backends/index.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+3. **Joint outreach**: outreach and promotion activities are performed together with other members of the federation, significantly reducing effort to attract new users.
+4. **Shared account management**: credit usage is tracked in a central manner, commercial usage is compensated in an efficient manner.
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/backends/index.md:44]: I'm trying to say that a backend will have less overhead doing the credit management, and things like billing users.
Of course the efficiency will eventually depend on procedures that are put in place...

File: APIs/openEO/federation/openeo_federation.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+The usage of processing capacity in the federation is likely to lead to credit consumption.
+This usage will be reported in the [openEO algorithm plaza](../../../Applications/PlazaDetails/Reporting.qmd). 
+To support basic processing within the federation, external providers can allow a limited amount of credits, similar to the [free openEO credits](../../../Quotas.qmd) offered by CDSE.
+It is important to note that there is clear separation of your credits amongst the external providers.
+This means that using an external provider will not lead to a reduction in the free credits supplied by CDSE. 
+
+## Getting help
+
+We encourage users to use the [forum](https://forum.dataspace.copernicus.eu/), a place where service providers and users can meet, to ask questions about the federation.
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/openeo_federation.md:32]: depends a bit on the release timeline indeed

File: APIs/openEO/federation/backends/index.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (JanssenBrm) [reply APIs/openEO/federation/backends/index.md]: I've slightly rephrased the text, you can revert the commit if necessary.

File: APIs/openEO/federation/backends/collections.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (JanssenBrm) [reply APIs/openEO/federation/backends/collections.md:53]: Correct, maybe we need to make it more explicit to announce it to the user and/or explain how to do it. Announcing could mean that they just put a message on their website or documentation but that may not be sufficient?

[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/backends/collections.md]: added a section about how to announce things.
We may need to refine it, but would be good to actually create the necessary tags or categories on the forum, so that we can properly link to them.

File: APIs/openEO/federation/backends/collections.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+considered non-experimental.
+
+1. Collections need to indicate the key 'providers' that are responsible for ensuring access to the data and continuity in the case of active missions. The user may depend on the guarantees offered by these
+providers with respect to the properties (timeliness, completeness,...) of a specific collection. 
+The providers with role 'host' and 'producer' are mandatory.
+
+2. The collection description and extents needs to specify known limitations with respect to the original collection. For instance, if only a subset of the full archive is available, this should be indicated. Extents can be rough approximations to avoid requiring very detailed geometry in the metadata.
+
+3. Collections without an end time are assumed to be active missions. By default, 99% of items in these collections should be available within 48 hours after being published by the producer. This gives users a basic guarantee with respect to timeliness of products. 
+4. Collection metadata should be valid STAC metadata and must include all extensions in `stac_extensions`. Tools such as [STAC-validator](https://github.com/stac-utils/stac-validator) can indicate obvious issues.
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/backends/collections.md]: I added a seperate requirement for datacube extension. Other than that one, I could not yet identify other required extensions. (openEO has fewer requirements in that regards compared to 'regular' stac catalogs.)

File: APIs/openEO/federation/backends/collections.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+3. Full-archive collections, or collections that fully cover a specific region. This to avoid 'demo' collections that only offer a marginal part of the actual dataset.
+
+
+## Collection availability
+
+The collections are a key asset of the federation, and users need to know what they can expect from a certain collection.
+The federation relies on openEO collection metadata following the STAC specification to communicate this towards the user. Backend providers
+need to comply with these requirements for their collection:
+
+The 'experimental' flag from https://stac-extensions.github.io/version/v1.1.0/schema.json needs to be set on experimental collections.
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/backends/collections.md]: indeed, field exists only in 1.2.0, updated the sentence. (Direct link doesn't seem to be available.)

File: APIs/openEO/federation/backends/api.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
+# Federation API
+
+The general contract is the [openEO API](https://api.openeo.org) in the latest stable version of the 1.x branch.
+
+The aggregator that proxies the back-ends in the federation also implements the same API, but it also implements the "Federation Extension" (currently in draft state).
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/backends/api.md]: added link

File: APIs/openEO/federation/backends/index.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+## How to join the federation
+
+1. Perform a basic check if you believe to be compliant with the above requirements. Determine the collections you believe to be eligible.
+2. Contact the federation via CDSE support, showing your interest and detailing your offering.
+3. Integrate the CDSE identity provider: you will need to accept CDSE users on your openEO service.
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/backends/index.md]: not yet, they will need to contact cdse for this, which they do anyway in step 2.
Would be nice to have public docs on this however, but outside of openEO scope.

File: APIs/openEO/federation/backends/api.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/backends/api.md:20]: already updated some references, but still need to update graphic

File: APIs/openEO/federation/backends/api.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+Authentication is performed via the CDSE OIDC provider.
+
+### Credits
+
+The second criterium for authorization is based on credits that are available to a user. Credits allow the platform to limit the volume of data access and processing operations that a user can perform during a given time frame. The amount of available credits depends on the subscription.
+When the credit balance of a user goes below zero, processing operations can be blocked.
+
+### Aggregator rules
+
+Based on the subscription and available credits, the aggregator can implement these rules:
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/backends/api.md]: updated, removed rate limiting

File: APIs/openEO/federation/backends/collections.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+1. Collections should be complementary to the already available offer. This avoids collections with the same name, or collections with differents names but similar content.
+2. Only non-experimental collections
+3. Full-archive collections, or collections that fully cover a specific region. This to avoid 'demo' collections that only offer a marginal part of the actual dataset.
+
+
+## Collection availability
+
+The collections are a key asset of the federation, and users need to know what they can expect from a certain collection.
+The federation relies on openEO collection metadata following the STAC specification to communicate this towards the user. Backend providers
+need to comply with these requirements for their collection:
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/backends/collections.md]: I removed the ':', but can you rephrase question?

File: APIs/openEO/federation/backends/index.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+- By default or if unsure, it is probably experimental.
+- If a feature is new or it hasn't been used, it's experimental.
+- If still in doubt, consult with partner providers.
+
+If a non-experimental component exceeds the [error budget](https://sre.google/workbook/implementing-slos/), for instance when downtime exceeds the objective, 
+the provider is expected to stop working on new features and improve reliability, or to mark the component 
+as experimental. Reverting a 'stable' feature to 'experimental' should be considered a backwards incompatible change, requiring communication towards the user and proper consideration of the impact.
+
+::: tip Note
+To join the federation, it is required to (mostly) fulfill these requirements and document differences for users in the "[Federation Aspects and Known Issues](../index.md)".
```
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/backends/index.md]: I simply removed the part containing the link.

File: APIs/openEO/federation/openeo_federation.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdries) [reply APIs/openEO/federation/openeo_federation.md]: I tried to further improve. 
I'm wondering if we should mandate that all providers should come from EU countries, thus making sure that EU law applies.

[the remark this answers is not in the document]
Comment (JanssenBrm) [reply APIs/openEO/federation/openeo_federation.md]: Good question, maybe we can await the feedback from ESA on this? As providers are requested to create a ticket, we can still evaluate this on a case-by-case basis.

**Review by JanssenBrm — APPROVED** [review]

**Comment by JanssenBrm** [comment]
It looks like the design page contains some tips that are not generated properly:
https://eu-cdse.github.io/documentation/pr-preview/pr-486/APIs/openEO/federation/backends/design.html

**Comment by Pratichhya** [comment]
> @Pratichhya - Can you maybe merge the staging branch again in this repository? I get merge conflicts in doing so on the submodule directories and I don't have any permissions to checkout these repos.

@JanssenBrm sure, I can

**Comment by Pratichhya** [comment]
@JanssenBrm could you confirm if the expected changes in the PR is there?

**Comment by JanssenBrm** [comment]
@Pratichhya  - My last change seems to be there, but the build is failing now.

**Comment by Pratichhya** [comment]
Hi Bram, I noticed that too and just managed to replicate it in my local. It's giving a quarto rendering error, though; with a similar setting, I get a different error in staging, while works well in publish. I will try to resolve it as soon as possible.

**Comment by JanssenBrm** [comment]
> Hi Bram, I noticed that too and just managed to replicate it in my local. It's giving a quarto rendering error, though; with a similar setting, I get a different error in staging, while works well in publish. I will try to resolve it as soon as possible.

It seems to be related to the NOTE section in the Browser.qmd file (https://github.com/eu-cdse/documentation/commit/205341629e1a175c9ddcef4bbc9333ad5740431f)

**Comment by Pratichhya** [comment]
> It seems to be related to the NOTE section in the Browser.qmd file ([2053416](https://github.com/eu-cdse/documentation/commit/205341629e1a175c9ddcef4bbc9333ad5740431f))


That was a quick and easy fix 😅 thank you 😄 
However, I noticed that not all the content in staging is up to date as Publish so will have to check that too I guess.

**Comment by Pratichhya** [comment]
migrated the changes to PR #529.
FYI @JanssenBrm
